### PR TITLE
test: fuzz mdx parser

### PR DESCRIPTION
**Pull request by ChristianMurphy**
This adds `.ok()` to the `Result` to capture and ignore recoverable errors.

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
This may want to wait on #26, currently `panic!` gets triggered pretty quickly

File: fuzz/fuzz_targets/markdown.rs [hunk 1]
```diff
@@ -9,7 +9,15 @@ fuzz_target!(|data: &[u8]| {
             s,
             &markdown::Options::gfm()
         );
+        let _ = markdown::to_html_with_options(
+            s,
+            &markdown::Options {
+                parse: markdown::ParseOptions::mdx(),
+                ..Default::default()
+            }
+        ).ok();
```
Comment (wooorm) [inline fuzz/fuzz_targets/markdown.rs:18]: This is pretty useless: MDX when going to HTML.
The latter one is useful: parsing MDX to an AST

File: fuzz/fuzz_targets/markdown.rs [hunk 2]
```diff
@@ -9,7 +9,15 @@ fuzz_target!(|data: &[u8]| {
...
         let _ = markdown::to_mdast(s, &markdown::ParseOptions::default());
         let _ = markdown::to_mdast(s, &markdown::ParseOptions::gfm());
+        let _ = markdown::to_mdast(s, &markdown::ParseOptions::mdx()).ok();
```
Comment (wooorm) [inline fuzz/fuzz_targets/markdown.rs:21]: What does the `.ok()` add? I believe it’ll work fine without it: that it’s a result and doesn’t panic?

File: fuzz/fuzz_targets/markdown.rs (hunk 1 above)
Comment (ChristianMurphy) [reply fuzz/fuzz_targets/markdown.rs]: removed

File: fuzz/fuzz_targets/markdown.rs (hunk 2 above)
Comment (ChristianMurphy) [reply fuzz/fuzz_targets/markdown.rs]: My impression is/was that some errors are to be expected with MDX?
My goal is to capture/ignore known errors, and only look for unexpected `panic!`.

That understand could be incorrect though.
Is the goal with the `mdx` parser to never throw?

---

in the meantime removed `.ok`

Comment (ChristianMurphy) [reply fuzz/fuzz_targets/markdown.rs]: also related https://github.com/wooorm/markdown-rs/pull/26#issuecomment-1296280425

**Pull request merged**